src/openloco/ui/dropdown.cpp

```diff
@@ -97,32 +115,338 @@ namespace openloco::ui::dropdown
+                    }
+                }
+                else
+                {
+                    auto x = _windowDropdownOnpaintCellX * _dropdownItemWidth + self->x + 2;
+                    auto y = _windowDropdownOnpaintCellY * 3 * (_dropdownItemHeight / 2) + self->y + 1;
+
+                    if (!(self->colours[0] & colour::translucent_flag))
+                    {
+                        gfx::draw_rect(dpi, x, y, _dropdownItemWidth - 1, 0, colour::get_shade(self->colours[0], 3));
+                        gfx::draw_rect(dpi, x, y + 1, _dropdownItemWidth - 1, 0, colour::get_shade(self->colours[0], 7));
+                    }
+                    else
+                    {
+                        auto colour = _byte_50457A[colour::opaque(self->colours[0])] | (1 << 25);
+                        colour++;
+                        gfx::draw_rect(dpi, x, y, _dropdownItemWidth - 1, 0, colour);
+                        colour++;
+                        gfx::draw_rect(dpi, x, y + 1, _dropdownItemWidth - 1, 0, colour);
+                    }
+                }
+
+                _windowDropdownOnpaintCellX++;
+                if (_windowDropdownOnpaintCellX >= _dropdownColumnCount)
+                {
+                    _windowDropdownOnpaintCellX = 0;
+                    _windowDropdownOnpaintCellY++;
+                }
+            }
+        }
+
+        static void initEvents()
+        {
+            events.on_update = onUpdate;
+            events.draw = draw;
+        }
+
+        // 0x004CCF1E
+        static void open(gfx::point_t origin, gfx::ui_size_t size, colour_t colour)
+        {
+            auto window = WindowManager::createWindow(WindowType::dropdown, origin, size, window_flags::stick_to_front, &common::events);
+
+            window->widgets = common::widgets;
+
+            if (colour & colour::translucent_flag)
+            {
+                window->flags |= window_flags::transparent;
+            }
+
+            common::initEvents();
+
+            common::widgets[0].colour = colour::black;
+            window->colours[0] = colour;
+
+            _dropdownHighlightedIndex = -1;
+            _dropdownDisabledItems = 0;
+            _dropdownSelection = 0;
+            input::state(input::input_state::dropdown_active);
+        }
+
+        static void setColourAndInputFlags(colour_t* colour, uint8_t* flags)
+        {
+            if (*colour & colour::translucent_flag)
+            {
+                *colour = _byte_504619[colour::opaque(*colour)];
+                *colour = colour::translucent(*colour);
+            }
+
+            input::reset_flag(input::input_flags::flag1);
+            input::reset_flag(input::input_flags::flag2);
+
+            if (*flags & (1 << 7))
+            {
+                input::set_flag(input::input_flags::flag1);
+            }
+
+            *flags &= ~(1 << 7);
+
+            WindowManager::close(WindowType::dropdown, 0);
```

Comment:
remove these two from this function as this function is just for setting colour and input flags.

Reply:
Make sure they are reinserted back into their respective functions